File: SMS Multicast/app/src/main/java/com/yuzhiqiang/smsmulticast/CSVImporter.java
package com.yuzhiqiang.smsmulticast;

import com.opencsv.CSVReader;

import java.io.FileReader;
import java.io.Reader;
import java.io.StringReader;
import java.util.ArrayList;

public class CSVImporter {

	public static ArrayList<Message> fromFile(String filePath) throws Exception {
		final FileReader fileReader = new FileReader(filePath);
		return importCSV(fileReader);
	}

	public static ArrayList<Message> fromString(String CSVString) throws Exception {
		final StringReader stringReader = new StringReader(CSVString);
		return importCSV(stringReader);
	}

	private static ArrayList<Message> importCSV(Reader reader) throws Exception {
		final CSVReader csvReader = new CSVReader(reader);
		ArrayList<Message> list = new ArrayList<>();
		while (true) {
			String[] line = csvReader.readNext();
			if (line == null)
				break;
			if (line.length != 2)
				throw new Exception();
			final String phoneNumber = line[0];
			final String message = line[1];
			list.add(new Message(phoneNumber, message));
		}
		return list;
	}

}

File: SMS Multicast/app/src/main/java/org/yuzhiqiang/smsmulticast/CSVImporter.kt
package org.yuzhiqiang.smsmulticast

import com.opencsv.CSVReader
import java.io.FileReader
import java.io.Reader
import java.io.StringReader
import java.util.*

object CSVImporter {

    @Throws(Exception::class)
    fun fromFile(filePath: String): ArrayList<Message> {
        val fileReader = FileReader(filePath)
        return importCSV(fileReader)
    }

    @Throws(Exception::class)
    fun fromString(CSVString: String): ArrayList<Message> {
        val stringReader = StringReader(CSVString)
        return importCSV(stringReader)
    }

    @Throws(Exception::class)
    private fun importCSV(reader: Reader): ArrayList<Message> {
        val csvReader = CSVReader(reader)
        val list = ArrayList<Message>()
        while (true) {
            val line = csvReader.readNext() ?: break
            if (line.size != 2)
                throw Exception()
            val phoneNumber = line[0]
            val message = line[1]
            list.add(Message(phoneNumber, message))
        }
        return list
    }

}
